frontend/src/crosssection/components/CrossMap.vue

```diff
@@ -0,0 +1,199 @@
+/*
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+<template>
+  <div id="cross-map" class="map"/>
+</template>
+
+<script>
+import mapboxgl from 'mapbox-gl'
+import GestureHandling from '@geolonia/mbgl-gesture-handling'
+
+import {
+  DATABC_ROADS_SOURCE,
+  DATABC_CADASTREL_SOURCE,
+  WELLS_SOURCE_ID,
+  DATABC_ROADS_SOURCE_ID,
+  DATABC_CADASTREL_SOURCE_ID,
+  DATABC_ROADS_LAYER,
+  DATABC_CADASTREL_LAYER,
+  wellsBaseAndArtesianLayer,
+  WELLS_BASE_AND_ARTESIAN_LAYER_ID,
+  FOCUSED_WELLS_SOURCE_ID,
+  focusedWellsLayer,
+  FOCUSED_WELL_IMAGE_ID,
+  FOCUSED_WELL_ARTESIAN_IMAGE_ID,
+  FOCUSED_WELL_CLOSED_IMAGE_ID,
+  wellLayerFilter,
+  WELLS_SOURCE
+} from '../../common/mapbox/layers'
+import { LegendControl } from '../../common/mapbox/controls'
+import { createWellPopupElement } from '../../common/mapbox/popup'
+import { PulsingWellImage, PulsingArtesianWellImage, PulsingClosedWellImage } from '../../common/mapbox/images'
+import {CENTRE_LNG_LAT_BC, buildWellsGeoJSON } from '../../common/mapbox/geometry'
+
+import wellsAllLegendSrc from '../../common/assets/images/wells-all.svg'
+import wellsArtesianLegendSrc from '../../common/assets/images/wells-artesian.svg'
+import wellsClosedLegendSrc from '../../common/assets/images/wells-closed.svg'
+import { setupFeatureTooltips } from '../../common/mapbox/popup'
+
+const WELL_FEATURE_PROPERTIES_FOR_POPUP = [
```

Comment:
Part of me wishes there was a way we could ensure that these are well properties (like pulling them from an object or smth) but I don't think we can so w/e.